### PR TITLE
vary restore cache on loadtest_factor

### PR DESCRIPTION
**Pull request by dannyroberts**
to fix https://manage.dimagi.com/default.asp?263060

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/restore_caching.py [hunk 1]
```diff
@@ -26,6 +28,16 @@ def invalidate(self):
         get_redis_default_cache().delete(self.cache_key)
 
 
+@quickcache(['domain', 'user_id'])
```
Comment (emord) [inline corehq/ex-submodules/casexml/apps/phone/restore_caching.py]: Should this be quickcached for longer?

**Comment by dannyroberts** [comment]
Didn't want it merged right before we all went offline for the summit.

Main hesitations here are that this adds something to every restore request, but I've tried to make that very minimal and only affect domains with the `ENABLE_LOADTEST_USERS` flag set. Am I correct in assuming that on environments where it matters (e.g. ICDS), flag lookups are hard-coded so there's ~zero performance hit to checking one, and that for other environments adding a cache lookup is totally fine? The only other hesitation is that cache invalidation is often wrong in some counter-intuitive way, but I've tried to work this out logically and believe it should work.

File: corehq/ex-submodules/casexml/apps/phone/restore_caching.py (hunk 1 above)
Comment (dannyroberts) [reply corehq/ex-submodules/casexml/apps/phone/restore_caching.py]: The default is 5 hours. What do you think would be a reasonable timeout? 24 hours?

Comment (emord) [reply corehq/ex-submodules/casexml/apps/phone/restore_caching.py]: I think the default is 5 minutes? not 5 hours?

**Comment by emord** [comment]
> flag lookups are hard-coded so there's ~zero performance hit to checking one

This is only true if they use `always_enabled` or `always_disabled` or some environment string that we added for enikshay recently in corehq/toggles.py

File: corehq/ex-submodules/casexml/apps/phone/restore_caching.py (hunk 1 above)
Comment (dannyroberts) [reply corehq/ex-submodules/casexml/apps/phone/restore_caching.py]: Yeah silly me you're right. I'll make it the same timeout as restore cache

**Review by emord — APPROVED** [review]
This change looks good to me. Will let you decide if that performance is good enough, or if we need to change toggles

**Pull request merged**